crates/rpc/rpc-eth-api/src/helpers/fee.rs

```diff
@@ -153,8 +153,17 @@ pub trait EthFees: LoadFee {
 
                 // Also need to include the `base_fee_per_gas` and `base_fee_per_blob_gas` for the
                 // next block
-                base_fee_per_gas
-                    .push(last_entry.next_block_base_fee(self.provider().chain_spec()) as u128);
+                let last_header = self
+                    .provider()
+                    .header(&last_entry.header_hash)
+                    .map_err(Self::Error::from_eth_err)?
+                    .ok_or(EthApiError::HeaderNotFound(last_entry.header_hash.into()))?;
+                base_fee_per_gas.push(
+                    self.provider()
+                        .chain_spec()
+                        .next_block_base_fee(&last_header, last_entry.timestamp)
+                        .unwrap_or_default() as u128,
+                );
```

Comment:
hmm, this isn't ideal, what if we track the header itself in the entry? then we dont need to read this here from disk